sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusTokenCredentialHttpPolicy.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+/**
+ * Token credential policy for authenticating with Service Bus.
+ */
+class ServiceBusTokenCredentialHttpPolicy implements HttpPipelinePolicy {
+    private final TokenCredential tokenCredential;
+
+    /**
+     * Creates a new credential policy using the Token credential.
+     *
+     * @param tokenCredential Token credential to get access token from.
+     */
+    ServiceBusTokenCredentialHttpPolicy(TokenCredential tokenCredential) {
+        this.tokenCredential = tokenCredential;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        final String url = context.getHttpRequest().getUrl().toString();
+        return tokenCredential.getToken(new TokenRequestContext().addScopes(url)).flatMap(token -> {
+            context.getHttpRequest().getHeaders().put("Authorization", token.getToken());
+            return next.process();
```

Comment:
May be we can use netty library available https://netty.io/4.1/api/io/netty/handler/codec/http/HttpHeaderNames.html#AUTHORIZATION